domain/application/state/unit.go

```diff
@@ -499,6 +502,64 @@ WHERE  u.name = $getUnitMachineUUID.unit_name
 	return arg.MachineUUID, nil
 }
 
+// getUnitMachineIdentifiers gets the identifiers of the machine that a unit is
+// attached to.
+//
+// The following errors may be expected:
+// - [applicationerrors.UnitNotFound] when the unit identified by the uuid no
+// longer exists.
+// - [machineerrors.MachineNotFound] when the unit is not assigned to a machine.
+func (st *State) getUnitMachineIdentifiers(
+	ctx context.Context, tx *sqlair.TX, unitUUID coreunit.UUID,
+) (internalapplication.MachineIdentifiers, error) {
+	var (
+		input = entityUUID{UUID: unitUUID.String()}
+		dbVal machineIdentifiers
+	)
+
+	q := `
+SELECT (m.uuid, m.net_node_uuid, m.name) AS (&machineIdentifiers.*)
+FROM   unit u
+JOIN   machine AS m ON u.net_node_uuid = m.net_node_uuid
+WHERE  u.uuid = $entityUUID.uuid
+`
+
+	stmt, err := st.Prepare(q, input, dbVal)
+	if err != nil {
+		return internalapplication.MachineIdentifiers{}, errors.Capture(err)
+	}
+
+	exists, err := st.checkUnitExists(ctx, tx, unitUUID)
+	if err != nil {
+		return internalapplication.MachineIdentifiers{}, errors.Errorf(
+			"checking unit %q exists", unitUUID,
+		)
+	}
+
+	if !exists {
+		return internalapplication.MachineIdentifiers{}, errors.Errorf(
+			"unit %q does not exist", unitUUID,
+		).Add(applicationerrors.UnitNotFound)
+	}
+
+	err = tx.Query(ctx, stmt, input).Get(&dbVal)
+	if errors.Is(err, sqlair.ErrNoRows) {
+		// While we expect the caller had validated this statement we can still
+		// provide a more helpful error message than sql error no rows.
+		return internalapplication.MachineIdentifiers{}, errors.Errorf(
+			"unit %q is not assigned to a machine in the model", unitUUID,
+		).Add(machineerrors.MachineNotFound)
```

Comment:
This should be a `applicationerrors.UnitMachineNotAssigned`.

Reply:
Was copying the existing logic we had :)